SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/ReferenceCountedLockingPolicy.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LoadBalancing/ReferenceCountedLockingPolicy.h"
+#include "GameFramework/Actor.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "EngineClasses/SpatialPackageMapClient.h"
+#include "Interop/SpatialStaticComponentView.h"
+#include "Schema/AuthorityIntent.h"
+
+DEFINE_LOG_CATEGORY(LogReferenceCountedLockingPolicy);
+
+ActorLockToken UReferenceCountedLockingPolicy::AcquireLock(const AActor* Actor)
```

Comment:
As far as I remember, `check` is removed in release builds, so `AcquireLock` will quietly return invalid value.

Reply:
This should only happen if devs have called AcquireLock without having called CanAcquireLock which is bad but I agree this would be an annoying thing to run into as a dev who didn't know better. Do you think we should log an error as well as checking? @samiwh ?

Reply:
Will make it so that you can call it with something you don't have authority over and it returns an invalid token and prints and error.